stats/views.py

```diff
@@ -27,8 +27,33 @@
 from .models import Scenario, ScenarioLanguage
 from .utils import get_label_properties_from_cache, prepare_label_cache
 
+from django.shortcuts import render
+from .forms import CaptchaForm
+import time
 
-class ScenarioList(FilterView):
+
+class LimitedFreeAccessMixin(AccessMixin):
+    """Verify that the current user is either authenticated or has successfully completed captcha test."""
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated or self.temporary_access_valid(request):
+            return super().dispatch(request, *args, **kwargs)
+        else:
+            # Redirect user to the captcha form page
+            return HttpResponseRedirect("/stats/captcha/")
+
+    def temporary_access_valid(self, request):
+            """
+            Check the validity of temporary access. Return true if the user has been granted temporary access. In this function, the access will be invalidated once it has passed the maximum time limit.
+            """
+            # Check if captcha has been completed properly
+            lack_captcha = request.session.get('succeed-captcha') is None or not request.session.get('succeed-captcha')
```

Comment:
You can simplify this a bit, `get()` has a second parameter with the default if the field is non-existent.

Reply:
Nice... thx :)